functions/go/main.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha512"
+	"encoding/hex"
+	"fmt"
+	"os"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+)
+
+func main() {
+	lambda.Start(handler)
+}
+
+func handler(r events.APIGatewayProxyRequest) (*events.APIGatewayProxyResponse, error) {
+	var (
+		webhookSecret = os.Getenv("WEBHOOK_SECRET")
+
+		timestamp = r.Headers["x-timestamp"]
+		nonce     = r.Headers["x-nonce"]
+		webhookID = r.Headers["x-webhook-id"]
+		gotHash   = r.Headers["x-signature"]
```

Comment:
interestingly these need to be lowercase.. for some reason, the request headers are all lowercased (might be due to some transformations with the AWS lambda library) -- but I can confirm we send these as `X-Timestamp`, etc..